datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -325,6 +327,75 @@ impl<'a> ConstEvaluator<'a> {
     }
 }
 
+/// Combine multiple OR expressions into a single IN list expression if possible
```

Comment:
Can we put it into a new file?
Because there will be more optimization about `expression`, which will make this file very long.

Reply:
Sure we can @jackwener